### ISSUE TITLE
Add "password forgotten" feature

### ISSUE THREAD

**Issue by leepeuker** [leepeuker/movary#47]
Currently the only way to help a user who has forgotten his/her password is to set a new password via cli command.

Todo:
- [x] add smtp mailer logic
- [ ] add "Forgot Password" button to login page
- [ ] create "Password reset" page, only with valid token
- [ ] send link to password reset page on "Forgot Password" button press
   - [ ] create and store password reset token in backend
   - [ ] must have an expiration time
 - [ ] Extend users setttings page with new section "Password resets" with two buttons:  "List Resets" and "Create Resets"

**Comment by JVT038** [comment]
What do you think about adding a column in the users table called 'reset_token_jwt' or something and storing a JWT token there, containing the expiry date (probably like 10 - 15 minutes), the user ID, and a random token?

The user will be sent an email with an URL that looks like `localhost/reset-password?token=<reset token>&id=<user id>` and when the user visits the page, Movary will check if the user id exists, if the user has a reset password JWT stored in their row, if the reset token corresponds with the stored jwt token and if the jwt has expired or not.

If all the conditions are met, the user will be allowed to view the reset password page and reset the password. If not, the user will just be redirected to the login page / home page with an error code or something.

**Comment by leepeuker** [comment]
Yes that is how I wanted to do this, except that I would have used a "simple" random token, I see no need for JWT here.

**Comment by JVT038** [comment]
Jwt supports an expiry date, which allows us to just use that for the expiry of the reset token.
This way, we don't need an entire separate table with a separate column dedicated to the expiry date of the reset tokens.
